java/src/main/native/src/TableJni.cpp

```diff
@@ -1323,6 +1175,118 @@ JNIEXPORT jlongArray JNICALL Java_ai_rapids_cudf_Table_readCSV(
+    jstring j_false_value, jobject j_buffer) {
+  JNI_NULL_CHECK(env, j_column_names, "column name array cannot be null", 0);
+  JNI_NULL_CHECK(env, j_row_delimiter, "row delimiter cannot be null", 0);
+  JNI_NULL_CHECK(env, j_field_delimiter, "field delimiter cannot be null", 0);
+  JNI_NULL_CHECK(env, j_null_value, "null representation string cannot be itself null", 0);
+  JNI_NULL_CHECK(env, j_buffer, "output buffer cannot be null", 0);
+
+  try {
+    cudf::jni::auto_set_device(env);
+
+    auto data_sink = std::make_unique<cudf::jni::jni_writer_data_sink>(env, j_buffer);
+
+    auto const n_column_names = cudf::jni::native_jstringArray{env, j_column_names};
+    auto const column_names = n_column_names.as_cpp_vector();
+
+    auto const line_terminator = cudf::jni::native_jstring{env, j_row_delimiter};
+    auto const na_rep = cudf::jni::native_jstring{env, j_null_value};
+    auto const true_value = cudf::jni::native_jstring{env, j_true_value};
+    auto const false_value = cudf::jni::native_jstring{env, j_false_value};
+
+    auto options = cudf::io::csv_writer_options::builder(cudf::io::sink_info{data_sink.get()},
+                                                         cudf::table_view{})
+                       .names(column_names)
+                       .include_header(static_cast<bool>(include_header))
+                       .line_terminator(line_terminator.get())
+                       .inter_column_delimiter(j_field_delimiter)
+                       .na_rep(na_rep.get())
+                       .true_value(true_value.get())
+                       .false_value(false_value.get())
+                       .build();
+
+    return ptr_as_jlong(new cudf::jni::io::csv_chunked_writer{options, data_sink});
+  }
+  CATCH_STD(env, 0);
+}
+
+JNIEXPORT void JNICALL Java_ai_rapids_cudf_Table_writeCSVChunkToBuffer(JNIEnv *env, jclass,
+                                                                       jlong j_writer_handle,
+                                                                       jlong j_table_handle) {
+  JNI_NULL_CHECK(env, j_writer_handle, "writer handle cannot be null.", );
+  JNI_NULL_CHECK(env, j_table_handle, "table handle cannot be null.", );
+
+  auto const table = reinterpret_cast<cudf::table_view *>(j_table_handle);
+  auto writer = reinterpret_cast<cudf::jni::io::csv_chunked_writer *>(j_writer_handle);
+
+  try {
+    cudf::jni::auto_set_device(env);
+    writer->write(*table);
+  }
+  CATCH_STD(env, );
+}
+
+JNIEXPORT void JNICALL Java_ai_rapids_cudf_Table_writeCSVToBufferEnd(JNIEnv *env, jclass,
+                                                                     jlong j_writer_handle) {
+  JNI_NULL_CHECK(env, j_writer_handle, "writer handle cannot be null.", );
+
+  using cudf::jni::io::csv_chunked_writer;
+  auto writer =
+      std::unique_ptr<csv_chunked_writer>{reinterpret_cast<csv_chunked_writer *>(j_writer_handle)};
```

Comment:
The `writer` in line 1280 can just be a pointer similar to line 1266.

Reply:
It can't, actually. The writer needs to be deleted, because it's allocated on heap.